src/applications/soundDetectionDemo/soundClassification/src/vSoundClassification.cpp

```diff
@@ -0,0 +1,358 @@
+            // Add the tone name inside the circle
+            cv::putText(image_tones_classification, tones_output_neurons_names[i], cv::Point(margin + (i * slot_size) + 5, 5 * (image_tones_classification.rows/6)),cv::FONT_HERSHEY_PLAIN, 0.5, cv::Vec3b(0, 0, 0), 0.5);
+            // ----------------------------------------------------------------------------
+
+            // ----------------------------------------------------------------------------
+            //                          HISTOGRAM DRAW
+            // First, get the normalized value of the neuron activity
+            float normalized_histogram_value = (tones_histogram[i] - min_value) * 100.0 / (max_value - min_value);
+
+            // Define left top point
+            float rec_x_top_left = margin + (i * slot_size) + 10;
+            // The y value is the bottom reference value minus the normalized value (the width of the histogram bar)
+            float rec_y_top_left = 4 * (image_tones_classification.rows/6) - normalized_histogram_value;
+            cv::Point p1(rec_x_top_left, rec_y_top_left);
+
+            // Define right bottom point
+            // X value is the same as left top point plus the diameter of the circle (radius * 2)
+            float rec_x_bottom_right = rec_x_top_left + (radius * 2);
+            float rec_y_bottom_right = 4 * (image_tones_classification.rows/6);
+            cv::Point p2(rec_x_bottom_right, rec_y_bottom_right);
+
+            // Create the rectangle
+            cv::rectangle(image_tones_classification, p1, p2, c);
+            // Add the normalized value to the rectangle
+            cv::putText(image_tones_classification, std::to_string((int)normalized_histogram_value), cv::Point(margin + (i * slot_size) + 10, 4 * (image_tones_classification.rows/6)),cv::FONT_HERSHEY_PLAIN, 0.8, cv::Vec3b(255, 255, 255), 0.5);
+            // ----------------------------------------------------------------------------
+        }
+        
+        // If debug flag is enabled
+        if(is_debug_flag == true){
+            // Print the winner
+            yInfo() << "Winner neuron is: " << pos_max_value;
+        }
+
+        // Show the image
+        cv::imshow("tones_out_result", image_tones_classification);
+        cv::waitKey(10);
+         
+        return Thread::isRunning();
+    }
+
+    // Asynchronous thread run forever
+    void run()
+    {
+        // YARP timestamp
+        Stamp yarpstamp;
+
+        // Output queue: type AE (not used)
+        deque<AE> out_queue;
+        // Output event: type AE (not used)
+        AE out_event;
+
+        // Raw address received from the input port
+        int address = 0;
+
+        // Initialize the short term memory deque
+        for(int i = 0; i < pure_tones_short_term_memory_size; i++) {
+            pure_tones_short_term_memory.push_back(0);
+        }
+
+        // Forever...
+        while(true) {
+
+            const vector<AE> * q = input_port.read(yarpstamp);
```

Comment:
it doesn't seem to me the best way to be handling this data if you are always using the `_coded_data` field. To me it means you are coding data in an event, in a way that that event type isn't meant to be used, this can only lead to errors when other people try to understand what you are doing.